lib/client-assets.php

```diff
@@ -703,10 +703,6 @@ function gutenberg_editor_scripts_and_styles( $hook ) {
 		gutenberg_get_rest_link( $post_to_edit, 'about', 'edit' ),
 	);
 
-	if ( ! $is_new_post ) {
```

Comment:
Preloading all revisions and exposing them in the Script tag was the main root cause of performance issues reported in #3167. Kudos to @pento for finding out 👍

Reply:
But why?

Reply:
See my description:

> This code preloads API response for revisions endpoint, which fetches all existing post revisions and exposes them in the <script /> tag. When you have let's say 50 revisions, all of them are loaded from the database and exposed in the HTML code leading to the big size of the HTML and increases processing time to process all objects.

And more technical comment from @pento:

> Loading all the revisions was running pretty much all of the code that would be run if the posts were being loaded to display on the front end. ‘the_content’ was the biggest problem, but everything added up.

Reply:
Sorry, definitely overlooked this 👍